include/mqtt/client.hpp

```diff
@@ -371,13 +372,26 @@ class client : public endpoint<std::mutex, std::lock_guard, PacketIdBytes> {
         set_keep_alive_sec_ping_ms(keep_alive_sec, keep_alive_sec * 1000 / 2);
     }
 
+
     /**
      * @brief Connect to a broker
      * Before calling connect(), call set_xxx member functions to configure the connection.
-     * @param func finish handler that is called when the session is finished
+     * @param session_life_keeper the passed object lifetime will be kept during the session.
      */
     void connect(any session_life_keeper = any()) {
-        connect(v5::properties{}, force_move(session_life_keeper));
+        boost::system::error_code ec;
+        connect(ec, force_move(session_life_keeper));
+        if (ec) throw ec;
```

Comment:
Is boost::system::error_code designed to be thrown? Does it inherit from std::exception?

Reply:
It should be something like

```cpp
        if (ec) throw boost::system::system_error(ec, "connect");
```

The current implementation could lose location information (2nd parameter of the system_error()). So I will fix to preserve the original information.